local/compose/run.go

```diff
@@ -0,0 +1,250 @@
+		Stream: true,
+		Stdin:  true,
+		Stdout: true,
+		Stderr: true,
+		Logs:   true,
+	})
+	if err != nil {
+		return err
+	}
+	defer cnx.Close()
+
+	stdout := convert.ContainerStdout{HijackedResponse: cnx}
+	stdin := convert.ContainerStdin{HijackedResponse: cnx}
+
+	readChannel := make(chan error, 10)
+	writeChannel := make(chan error, 10)
+
+	go func() {
+		_, err := io.Copy(os.Stdout, cnx.Reader)
+		readChannel <- err
+	}()
+
+	go func() {
+		_, err := io.Copy(stdin, os.Stdin)
+		writeChannel <- err
+	}()
+
+	go func() {
+		<-ctx.Done()
+		stdout.Close() //nolint:errcheck
+		stdin.Close()  //nolint:errcheck
+	}()
+
+	// start container
+	err = s.apiClient.ContainerStart(ctx, container, apitypes.ContainerStartOptions{})
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case err := <-readChannel:
+			return err
+		case err := <-writeChannel:
+			return err
+		}
+	}
+}
+
+func updateOneOffServiceConfig(service *types.ServiceConfig, projectName string, opts compose.RunOptions) {
+	if len(opts.Command) > 0 {
+		// custom command to run
+		service.Command = opts.Command
+	}
+	//service.Environment = opts.Environment
+	slug := moby.GenerateRandomID()
+	service.Scale = 1
+	service.ContainerName = fmt.Sprintf("%s_%s_run_%s", projectName, service.Name, moby.TruncateID(slug))
+	service.Labels = types.Labels{
+		"com.docker.compose.slug":   slug,
+		"com.docker.compose.oneoff": "True",
+	}
+	service.Tty = true
+	service.StdinOpen = true
+}
+
+func (s *composeService) ensureRequiredServices(ctx context.Context, project *types.Project, service types.ServiceConfig) error {
+	requiredServices := getDependencyNames(project, service, func() []string {
+		return service.GetDependencies()
+	})
+	if len(requiredServices) > 0 {
+		// dependencies here
+		services, err := project.GetServices(requiredServices)
+		if err != nil {
+			return err
+		}
+		project.Services = services
+		err = s.ensureImagesExists(ctx, project)
+		if err != nil {
+			return err
+		}
+
+		err = InDependencyOrder(ctx, project, func(c context.Context, svc types.ServiceConfig) error {
+			return s.ensureService(c, project, svc)
+		})
+		if err != nil {
+			return err
+		}
+		return s.Start(ctx, project, nil)
+	}
+	return nil
+}
+
+func (s *composeService) ensureRequiredNetworks(ctx context.Context, project *types.Project, service types.ServiceConfig) error {
+	networks := getDependentNetworkNames(project, service)
+	for k, network := range project.Networks {
+		if !contains(networks, network.Name) {
+			continue
+		}
+		if !network.External.External && network.Name != "" {
+			network.Name = fmt.Sprintf("%s_%s", project.Name, k)
+			project.Networks[k] = network
+		}
+		network.Labels = network.Labels.Add(networkLabel, k)
+		network.Labels = network.Labels.Add(projectLabel, project.Name)
+		network.Labels = network.Labels.Add(versionLabel, ComposeVersion)
+
+		err := s.ensureNetwork(ctx, network)
+		if err != nil {
+			return err
+		}
```

Comment:
Need to find a better way to avoid duplicating all this. (Happy to pair program on this a bit if you like)